File: sdks/js/abi/abi.ts
export const POLYPLUG_ABI_VERSION: number = 1;

export function fnv1a_64(data: &[u8]): bigint {}

export function contract_id(name: &str, major: number): bigint {}

export function bundle_id(name: &str): bigint {}

export function host_contract_id(name: &str, major: number): bigint {}

export function plugin_contract_id(name: &str, major: number): bigint {}

File: sdks/lua/abi/abi.lua
local ffi = require("ffi")
local M = {}

M.POLYPLUG_ABI_VERSION = ffi.cast("uint32_t", 1)
local function fnv1a_64(&[u8] data) end

local function contract_id(&str name, uint32_t major) end

local function bundle_id(&str name) end

local function host_contract_id(&str name, uint32_t major) end

local function plugin_contract_id(&str name, uint32_t major) end

return M
